README.md

```diff
@@ -13,3 +13,32 @@
   complexity suggests a different time.
 
 Add your answers to this markdown file.
+
+Bullet 1
+
+1. Asymptotic analysis is theoretical, as it describes algorithm behavior with a set of unspecific inputs.  It doesn’t always take into account the size of the input it is dealing with.  Actual performance is empirical, as it describes algorithm behavior with a specific set of inputs.  The two are different, as actual performance depends on constants and machine performance, whereas asymptotic analysis ignores those factors.
+
+2. Actual performance is affected by hardware, software, and the characteristics of the input data, whereas asymptotic analysis is primarily affected by the input size.
+
+3. In addition to ignoring constants, asymptotic analysis also ignores lower-order terms, which can be misleading when compared to actual performance since when lower-order terms are accounted for, they can have a decent effect on the runtime.
+
+Bullet 2
+
+The time complexity depends on if the binary search tree is balanced or not.  If it is balanced, it has a best case of O(1) and an average and worst case of O(log(n)).  If it isn’t balanced, it has a best case of O(1), and an average and worst case of O(n).
+
+For a balanced binary tree, finding the same element in a tree of 10,000 elements takes about 6.65 seconds in the worst case, which comes from multiplying 5 by log2(10,000)/log2(1,000).  For an unbalanced binary tree, finding the same element in a tree of 10,000 elements takes about 50 seconds in the worst case, which comes from multiplying 5 by 10,000/1,000. 
+
+Bullet 3
+
+1. It could have taken much longer because the tree was very unbalanced, meaning the algorithm has to go through most, if not all of the nodes.  This causes a large difference, as with a very unbalanced tree, the search time can approach O(n) instead of O(logn).  This alone doesn’t fully account for the large jump from 5 seconds to 100 seconds.  Additional overheads such as memory issues or implementation inefficiencies combined with the very unbalanced tree could make the time take significantly longer.
```

Comment:
You're estimating how long it would take if the tree were imbalanced above and you're getting a completely different number. So this can't be the reason, as you found yourself.

Reply:
I'm not sure if I understand what you mean by me getting a completely different number.

Reply:
Oh, that's right.  But wouldn't the combination of an unbalanced binary tree and additional overheads like memory issues or implementation inefficiencies cause the runtime to be considerably longer?